boa_engine/src/object/shape/shared_shape/mod.rs

```diff
@@ -192,6 +192,8 @@ impl SharedShape {
             if let Some(inner) = shape.upgrade() {
                 return Self { inner };
             }
+
+            self.forward_transitions().prune_prototype_transitions();
```

Comment:
I may be missing something here, but what is the reason for the explicit calls to prune here, when `insert_property` and `insert_prototype` are always called afterwards and contain the prune code too?